extensions/spark/kyuubi-spark-lineage/src/main/scala/org/apache/kyuubi/plugin/lineage/helper/SemanticVersion.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.plugin.lineage.helper
+
+/**
+ * Encapsulate a component (Kyuubi/Spark/Hive/Flink etc.) version
+ * for the convenience of version checks.
+ */
+case class SemanticVersion(majorVersion: Int, minorVersion: Int) {
```

Comment:
hmm, before introducing this, the version comparison seems to be very simple. Now, plenty copies of this class..

Reply:
> hmm, before introducing this, the version comparison seems to be very simple. Now, plenty of copies of this class.

In this plugin we only use isSparkVersionAtMost method, do I need to simply reimplement a method instead of copying `SemanticVersion`?